code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPUtils.java

```diff
@@ -32,7 +34,7 @@
      * and resubmit it, without locking them out of cattle.
      */
     public static void validateConfig(LDAPConstants ldapConfig) {
-        if (ldapConfig.getEnabled() == null || !ldapConfig.getEnabled()) {
+        if (ldapConfig.getEnabled() == null || !ldapConfig.getEnabled() || SecurityConstants.SECURITY.get()) {
```

Comment:
Long explanation :), bear with me ;)

To configure and enable access control, UI calls the access control resource API's in following order:
Example: For LDAP active directory provider the API's and order is:
1) /v1/ldapconfig with ldap configuration params and enabled = "false"
2) /v1/token This API uses the above set config and tests the login for a test user and if login is successful a token is generated and returned. 
3) If step2 is successful, access control can be "enabled" by calling /v1/ldapconfig API again with "enabled" = true. Internally this sets the flag SecurityConstants.SECURITY as true indicating rancher accesscontrol is set.

Now the above code in question validateConfig(ldapConfig), tests that the config provided is correct by connecting to the active directory provided, before turning the access control on rancher ON. This validation is done only when step3 is performed (/v1/ldapconfig API with "enabled" = true)

Now! about the additional check I added: 
When "restricted" accessmode support is added(what the PR is about), the Admin can set the "allowedIdentities" anytime after the accessControl is turned ON.
The same API (/v1/ldapconfig API with "enabled" = true and allowedIdentities = [] and accessMode = "restricted") is called to update the ldapconfig. 

The validation above can not be done in this case, since this /v1/ldapconfig call does not carry the admin secret (since this step can happen anytime after access control is turned ON). 
Thus to bypass the validation, checking the SecurityConstants.SECURITY.get() tells us that access control is already ON, we need not validate the config again.